src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBase.cs

```diff
@@ -0,0 +1,155 @@
+	{
+		MauiApp _mauiApp;
+		IServiceProvider _servicesProvider;
+		IMauiContext _mauiContext;
+		bool _isCreated;
+
+		public void EnsureHandlerCreated(Action<MauiAppBuilder> additionalCreationActions = null)
+		{
+			if (_isCreated)
+			{
+				return;
+			}
+
+			_isCreated = true;
+
+
+			var appBuilder = ConfigureBuilder(MauiApp.CreateBuilder());
+
+			additionalCreationActions?.Invoke(appBuilder);
+
+			appBuilder.Services.TryAddSingleton<IDispatcherProvider>(svc => TestDispatcher.Provider);
+			appBuilder.Services.TryAddScoped<IDispatcher>(svc => TestDispatcher.Current);
+			appBuilder.Services.TryAddSingleton<IApplication>((_) => new ApplicationStub());
+
+			_mauiApp = appBuilder.Build();
+			_servicesProvider = _mauiApp.Services;
+
+			_mauiContext = new ContextStub(_servicesProvider);
+		}
+
+		protected virtual MauiAppBuilder ConfigureBuilder(MauiAppBuilder mauiAppBuilder)
+		{
+			return mauiAppBuilder;
+		}
+
+		protected IMauiContext MauiContext
+		{
+			get
+			{
+				EnsureHandlerCreated();
+				return _mauiContext;
+			}
+		}
+
+		protected Task SetValueAsync<TValue, THandler>(IView view, TValue value, Action<THandler, TValue> func)
+			where THandler : IElementHandler, new()
+		{
+			return InvokeOnMainThreadAsync(() =>
+			{
+				var handler = CreateHandler<THandler>(view);
+				func(handler, value);
+			});
+		}
+
+		protected THandler CreateHandler<THandler>(IElement view, IMauiContext mauiContext = null)
+			where THandler : IElementHandler, new()
+			=> CreateHandler<THandler, THandler>(view, mauiContext);
+
+
+		protected void InitializeViewHandler(IElement element, IElementHandler handler, IMauiContext mauiContext = null)
+		{
+			mauiContext ??= MauiContext;
+			handler.SetMauiContext(mauiContext);
+			handler.SetVirtualView(element);
+			element.Handler = handler;
+
+			if (element is IView view && handler is IViewHandler viewHandler)
+			{
+#if ANDROID
+				// If the Android views don't have LayoutParams set, updating some properties (e.g., Text)
+				// can run into issues when deciding whether a re-layout is required. Normally this isn't
+				// an issue because the LayoutParams get set when the view is added to a ViewGroup, but 
+				// since we're not doing that here, we need to ensure they have LayoutParams so that tests
+				// which update properties don't crash. 
+
+				var aView = viewHandler.PlatformView as Android.Views.View;
+				if (aView.LayoutParameters == null)
+				{
+					aView.LayoutParameters =
+						new Android.Views.ViewGroup.LayoutParams(
+							Android.Views.ViewGroup.LayoutParams.WrapContent,
+							Android.Views.ViewGroup.LayoutParams.WrapContent);
+				}
+
+				var size = view.Measure(view.Width, view.Height);
+				var w = size.Width;
+				var h = size.Height;
+#elif IOS || MACCATALYST
+				var size = view.Measure(double.PositiveInfinity, double.PositiveInfinity);
+				var w = size.Width;
+				var h = size.Height;
+
+				if (double.IsPositiveInfinity(w))
```

Comment:
@hartez I added this code because all of the `ScrollView` tests started failing due to `view.Measure` returning infinite size.

I'm wondering if this is actually a bug with `ScrollViewHandler`. 

The `GetDesiredSize` code inside `ScrollViewHandler` just returns the constraints back if the `ScrollView` has no content (which makes sense)

https://github.com/dotnet/maui/blob/main/src/Core/src/Handlers/ScrollView/ScrollViewHandler.iOS.cs#L226-L229

I wonder if for this scenario the `ScrollViewHandler` on iOS should return a size of zero if the content is null and the constraints are infinite. 

Something like

```C#
if (platformView == null || virtualView == null || virtualView.PresentedContent == null)
{
     if (double.IsPositiveInfinity(widthConstraint))
          widthConstraint = 0;
          
     if (double.IsPositiveInfinity(heightConstraint))
          heightConstraint= 0;
          
     return new Size(widthConstraint, heightConstraint);
}
```